pkg/operator/configobservation/encryption/observe_encryption.go

```diff
@@ -0,0 +1,69 @@
+package encryption
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	encryptionConfFilePath = "/etc/kubernetes/static-pod-resources/secrets/encryption-config/encryption-config"
+	encryptionConfSecret   = "encryption-config"
+)
+
+type SecretLister interface {
+	SecretLister() corev1listers.SecretLister
+}
+
+func NewEncryptionObserver(targetNamespace string, encryptionConfigPath []string) configobserver.ObserveConfigFunc {
+	return func(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
strange to return a func this big

Reply:
> strange to return a func this big

I did not want to create a new struct just to close over the inputs.